crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -582,6 +583,14 @@ impl ExecutorState {
             // - `update_archetype_component_access` has been called for system.
             let valid_params = unsafe { system.validate_param_unsafe(world) };
             if !valid_params {
+                error_handler(
+                    SystemParamValidationError::System.into(),
+                    ErrorContext::System {
+                        name: system.name(),
+                        last_run: system.get_last_run(),
+                    },
+                );
```

Comment:
It seems `ErrorContext` is enough to tell whether we're working with a system/observer, etc. do we need to duplicate this in `SystemParamValidationError`?